src/core/load-fragment.c

```diff
@@ -5423,6 +5423,87 @@ int config_parse_ip_filter_bpf_progs(
         return 0;
 }
 
+int config_parse_restrict_network_interfaces(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        CGroupContext *c = data;
+        bool is_allow_rule = true;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                c->restrict_network_interfaces = set_free_free(c->restrict_network_interfaces);
```

Comment:
in new code we want to avoid set_free_free() ideally, and use plain set_free() instead. And instead pass the destructor to use as part of the "hash_ops" structure passed to set_new(). Given that you want to store a string here, the right hash_ops to use is `string_hash_ops_free`

Reply:
Eh, I did all that mess of loops and gotos because I didn't know that `string_hash_ops_free` will do the magic of handling strings for me. Thanks for pointing this out.